internal/store/segment_test.go

```diff
@@ -0,0 +1,157 @@
+package store_test
+
+import (
+	"context"
+	"github.com/bidon-io/bidon-backend/internal/admin"
+	"testing"
+
+	"github.com/bidon-io/bidon-backend/internal/store"
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestSegmentRepo_List(t *testing.T) {
+	tx := db.Begin()
+	defer tx.Rollback()
+
+	repo := &store.SegmentRepo{DB: tx}
+
+	segments := []admin.SegmentAttrs{
+		{
+			Name:        "Country Segment",
+			Description: "Desc",
+			AppID:       1,
+			Filters:     []admin.SegmentFilter{{Type: "country", Name: "country", Operator: "in", Values: []string{"US", "UK"}}},
+			Enabled:     func() *bool { b := true; return &b }(),
```

Comment:
I think it would be better to define a generic helper function instead

```go
func ptr[T any](t T) *T {
    return &t
}
```

```suggestion
			Enabled:    ptr(false),
```